### PR TITLE
DM-43334: Add dmL1AstroErr/dmL2AstroErr metric

### PR DESCRIPTION
**Pull request by cmsaunders**

### CODE REVIEW

File: pipelines/matchedVisitQualityCore.yaml [hunk 1]
```diff
@@ -21,6 +21,12 @@ tasks:
       atools.stellarAstrometricRepeatability3.xValue: 3
       atools.stellarAstrometricRepeatability3.process.calculateActions.rms.annulus: 200
       atools.stellarAstrometricRepeatability3.process.calculateActions.rms.threshAD: 30
+      atools.stellarAstrometricSelfRepeatabilityRA: AstrometricRepeatability
+      atools.stellarAstrometricSelfRepeatabilityRA.level: 2
+      atools.stellarAstrometricSelfRepeatabilityRA.coordinate: 'RA'
+      atools.stellarAstrometricSelfRepeatabilityDec: AstrometricRepeatability
+      atools.stellarAstrometricSelfRepeatabilityDec.level: 2
+      atools.stellarAstrometricSelfRepeatabilityDec.coordinate: 'RA'
```
Comment (jeffcarlin) [inline pipelines/matchedVisitQualityCore.yaml:29]: 'RA' --> 'Dec'

**Comment by jeffcarlin** [comment]
I agree with your choice of implementation. I confirmed that the code runs fine and produces reasonable results (though note that you'll need to change "RA" to "Dec" in the matchedVisitCore pipeline). Just a few small comments, and I think this is good to go!

**Review by jeffcarlin — CHANGES_REQUESTED** [review]
I agree with your choice of implementation. I confirmed that the code runs fine and produces reasonable results (though note that you'll need to change "RA" to "Dec" in the matchedVisitCore pipeline). Just a few small comments, and I think this is good to go!

File: python/lsst/analysis/tools/atools/astrometricRepeatability.py [hunk 2]
```diff
@@ -265,3 +268,60 @@ def finalize(self):
...
+    def setDefaults(self):
+        super().setDefaults()
+        self.prep.selectors.bandSelector = BandSelector()
+
+        self.process.buildActions.perGroupStd = ConvertUnits()
+        self.process.buildActions.perGroupStd.inUnit = "degree"
+        self.process.buildActions.perGroupStd.outUnit = "marcsec"
+        self.process.buildActions.perGroupStd.buildAction = PerGroupStatistic()
+        self.process.buildActions.perGroupStd.buildAction.func = "std"
+        self.process.buildActions.perGroupStd.buildAction.buildAction.vectorKey = "coord_dec"
```
[the remark this answers is not in the document]
Comment (cmsaunders) [reply python/lsst/analysis/tools/atools/astrometricRepeatability.py:292]: Something has to be set, or the configuration validation fails. However, if I have this, I don't actually need the `else` part of the `if self.coordinate == "RA":.. ` statement in `finalize()`, so I took that out.

File: python/lsst/analysis/tools/atools/astrometricRepeatability.py [hunk 3]
```diff
@@ -265,3 +268,60 @@ def finalize(self):
             "AFx": f"{{band}}_AF{self.xValue}",
             "ADx": f"{{band}}_AD{self.xValue}",
         }
+
+
+class AstrometricRepeatability(AnalysisTool):
+    """Calculate the median position RMS of point sources."""
+
+    fluxType = Field[str](doc="Flux type to calculate repeatability with", default="psfFlux")
+    level = Field[int](doc="Level 1 or 2 data product (1 or 2)", default=1)
```
[the remark this answers is not in the document]
Comment (cmsaunders) [reply python/lsst/analysis/tools/atools/astrometricRepeatability.py:277]: Yes, good point! That should be more clear.

**Review by jeffcarlin — APPROVED** [review]
Looks good!

**Pull request merged**